rfcs/0000_project_typing_guidelines.md

```diff
@@ -177,8 +176,26 @@ def do_maths(input: ...) -> npt.NDArray[np.float64]:
     return result
 ```
 
-This might require the use of [`typing.cast()`](https://docs.python.org/3/library/typing.html#typing.cast) to satisfy the type checker.
+This might require the use of [`typing.cast()`](https://docs.python.org/3/library/typing.html#typing.cast) or explicit type hints on variable declarations to satisfy the type checker.
 
 The granularity SHOULD NOT be individual NumPy functions, but rather whole operations in our code where we are using multiple NumPy functions in a row. We're not trying to make a wrapper for poorly typed libraries, but making sure when we return things into our code, store values on objects, etc. that we're giving them useful types.
 
-This would be also good place to use nominal types to define _input_ argument types like quaternion tuples that have a particular order so that callers don't pass the wrong values into these external library functions. See the nominal type guidance above.
+```python
+# Don't do something this granular
+def get_worldbody(spec: MjSpec) -> MjsBody:
+    return spec.worldbody  # this might require `cast(MjsBody, spec.worldbody)`
+
+def some_other_method(...) -> None:
+    ...
+    worldbody = get_worldbody(spec)
+    ...
+
+# Instead, just declare the types in the other method where needed
+def some_other_method(...) -> MjsBody:
+   ...
+   worldbody: MjsBody = spec.worldbody
+   # More MuJoCo calls on worldbody that may need type hints
+   ...
+```
+
+This would be also good place to use newtypes to define _input_ argument types like quaternion tuples that have a particular order so that callers don't pass the wrong values into these external library functions. See [the nominal type guidance](#nominal-typing-should-be-used-to-name-concepts-using-types) above.
```

Comment:
```suggestion
This would also be a good place to use newtypes to define _input_ argument types like quaternion tuples that have a particular order so that callers don't pass the wrong values into these external library functions. See [the nominal type guidance](#nominal-typing-should-be-used-to-name-concepts-using-types) above.
```